src/ensembl/src/content/app/browser/browser-region-field/BrowserRegionField.tsx

```diff
@@ -0,0 +1,226 @@
+      setIsFormSubmitted(true);
+      props.validateRegion(getRegionInputWithRegion(regionFieldInput));
+    }
+  };
+
+  const changeLocation = (inputValue: string) => {
+    if (!isFormSubmitted) {
+      return;
+    }
+
+    props.changeBrowserLocation(
+      props.activeGenomeId as string,
+      getChrLocationFromStr(getRegionInputWithRegion(inputValue))
+    );
+  };
+
+  const changeFocusObject = (inputValue: string) => {
+    const activeEnsObjectId = `${
+      props.activeGenomeId
+    }:region:${getRegionInputWithRegion(inputValue)}`;
+
+    const params = {
+      genomeId: props.activeGenomeId,
+      focus: activeEnsObjectId,
+      location: getRegionInputWithRegion(inputValue)
+    };
+
+    props.replace(urlFor.browser(params));
+  };
+
+  const updateBrowser = () => {
+    if (!isFormSubmitted) {
+      return;
+    }
+
+    const [region, ,] = props.chrLocation;
+    const regionInput = regionFieldInput.includes(':')
+      ? regionFieldInput.split(':')[0]
+      : region;
+
+    if (regionInput === region) {
+      changeLocation(regionFieldInput);
+    } else {
+      changeFocusObject(regionFieldInput);
+    }
+  };
+
+  useEffect(() => () => props.resetRegionValidation(), []);
+
+  useEffect(() => {
+    if (props.regionFieldActive) {
+      const { regionValidationInfo, regionValidationLoadingStatus } = props;
+
+      if (regionValidationLoadingStatus !== LoadingState.LOADING) {
+        const errorMessages = getRegionFieldErrorMessages(regionValidationInfo);
+
+        if (errorMessages) {
+          setErrorMessages(errorMessages);
+        } else {
+          setRegionFieldInput('');
+          setErrorMessages(null);
+          setIsFormSubmitted(false);
+          props.toggleRegionFieldActive(false);
+
+          updateBrowser();
+        }
+      }
+    }
+  }, [props.regionValidationInfo]);
+
+  const regionFieldClassNames = classNames(styles.browserRegionField, {
+    [browserStyles.semiOpaque]: props.regionEditorActive
+  });
+
+  const inputClassNames = classNames(browserNavBarStyles.inputText, {
+    [browserNavBarStyles.errorText]: errorMessages
+  });
+
+  const buttonsClassNames = classNames(
+    browserNavBarStyles.browserNavBarButtons,
+    {
+      [browserNavBarStyles.browserNavBarButtonsVisible]: props.regionFieldActive
+    }
+  );
+
+  return (
+    <div className={regionFieldClassNames}>
+      {props.regionEditorActive ? (
+        <div className={browserStyles.browserOverlay}></div>
+      ) : null}
+      <form onSubmit={handleSubmit}>
+        <label className="show-for-large">Region or location</label>
+        <Input
+          type="text"
+          placeholder="0:1-1,000,000"
+          value={regionFieldInput}
+          onChange={changeRegionFieldInput}
+          onFocus={activateForm}
+          className={inputClassNames}
+        />
+        <span className={buttonsClassNames}>
+          <button type="submit" className="submitButton">
```

Comment:
There isn't any style associated with this button. Same in RegionEditor.

If the only purpose of this class is to find this element in the test, then perhaps `button[type="submit"]` will be enough to identify it?

Reply:
If I am not wrong, `jest` can't parse attributes properly. I tried all those approaches before settling with classes as they all ended up failing my test cases.

Reply:
Tried this again with quoting the attribute values with quotes e.g. `button[type="submit"]. That did work!